### PR TITLE
chore: add support for OSSF scorecard reporting

### PR DESCRIPTION
**Pull request by inigomarquinez**
## Main Changes

This pipeline will proactively report the status of the project (every day and when a push is done to `master` branch) including critical fields (CI-Tests, Contributors, Dependency-Update-Tool, Webhooks) that are missing while running via OSSF cron jobs.

### Changes related

> It's also possible that some repositories in your organization are already being [automatically tracked](https://github.com/ossf/scorecard/blob/main/docs/faq.md#can-i-preview-my-projects-score) by OpenSSF in this [CSV file](https://github.com/ossf/scorecard/blob/main/cron/internal/data/projects.csv) via weekly cronjob. One caveat: Automatically tracked projects do not include https://github.com/ossf/scorecard/issues/3438 in their analysis (CI-Tests,Contributors,Dependency-Update-Tool,Webhooks).
Source: [openssf-scorecard-monitor documentation](https://github.com/UlisesGascon/openssf-scorecard-monitor)

### Team discussion related

- Ref: https://github.com/expressjs/security-wg/issues/2
- Report: https://kooltheba.github.io/openssf-scorecard-api-visualizer/#/projects/github.com/pillarjs/encodeurl/commit/5f2205efe622fc7972c4e45ed7cbe7f8d652c299

### CODE REVIEW

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# policy, and support documentation.
+
+name: Scorecard supply-chain security
+
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
```
Comment (blakeembrey) [inline .github/workflows/scorecard.yml:12]: I'm a little worried about this, the package shouldn't need to be kept "up to date" re https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained.

**Review by blakeembrey — COMMENTED** [review]
Is there a reason we want to use the specific commit hashes instead of version numbers for every action?

**Comment by blakeembrey** [comment]
I found my answer here: https://github.com/ossf/scorecard/blob/d58bfb03aab496807d8489e09da9883928465f59/docs/checks.md#pinned-dependencies 😓 

Seems like a GitHub Actions "lock file" would be a nice feature to simplify future updates.

**Comment by blakeembrey** [comment]
A heads up, but I'm already seeing this:

> CodeQL Action v2 will be deprecated on December 5th, 2024. Please update all occurrences of the CodeQL Action in your workflow files to v3. For more information, see https://github.blog/changelog/2024-01-12-code-scanning-deprecation-of-codeql-action-v2/

**Comment by inigomarquinez** [comment]
Sorry I didn't see your messages before @blakeembrey. Thanks for reviewing!

**Comment by Queenshg87** [comment]
Need review i believe..

**Pull request merged**